fury/utils.py

```diff
@@ -1073,3 +1073,43 @@ def get_bounds(actor):
 
     """
     return actor.GetMapper().GetInput().GetBounds()
+
+
+def clip_overflow(textblock, width):
+    """Clips overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be clipped.
+    width : int
+        Required width of the clipped text.
+
+    Returns
+    -------
+    clipped text : str
+        Clipped version of the text.
+    """
+    original_str = textblock.message
+    start_ptr = 0
+    end_ptr = len(original_str)
+    prev_bg = textblock.have_bg
+    textblock.have_bg = False
+
+    if textblock.size[0] == width or textblock.size[0] <= width:
+        textblock.have_bg = prev_bg
+        return original_str
+
+    else:
+        while start_ptr < end_ptr:
```

Comment:
I wonder about the impact of this `while` for performance.  `clip_overflow` seems to be called often. I need to test it. As a temporary solution, I will be fine but we will have to find a better approach.

Reply:
The complexity of this is around `log2(length)`. The problem is each character has a different height and width.
Also the width of two characters within a string isn't equal to the sum of their individual widths.